### ISSUE TITLE
Hot reload button didn't perform hot reload

### ISSUE THREAD

**Issue by InMatrix** [flutter/flutter-intellij#1228]
## Steps to Reproduce

- Create a new Flutter project
- Run the app in an iPhone SE simulator 
- Change the theme color of the app
- Hit the Hot Reload button in IntelliJ

The console said "Reloaded 1 of 439 libraries in 305ms." But the color wasn't updated in the Simulator. 

This problem seems to only exist in IntelliJ. Hot Reload via command line worked. 

## Version info

[✓] Flutter (on Mac OS X 10.12.5 16F73, locale en-US, channel master)
    • Flutter at /Users/taodong/Code/flutter_repos/flutter
    • Framework revision 48427cb77a (3 hours ago), 2017-07-20 17:06:57 -0700
    • Engine revision 5fcfb995bb
    • Tools Dart version 1.25.0-dev.4.0

[✓] Android toolchain - develop for Android devices (Android SDK 25.0.2)
    • Android SDK at /Users/taodong/Library/Android/sdk
    • Platform android-24, build-tools 25.0.2
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_112-release-b06)

[✓] iOS toolchain - develop for iOS devices (Xcode 8.3.3)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 8.3.3, Build version 8E3004b
    • ios-deploy 1.9.1
    • CocoaPods version 1.2.0

[✓] Android Studio (version 2.3)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Java version OpenJDK Runtime Environment (build 1.8.0_112-release-b06)

[✓] IntelliJ IDEA Community Edition (version 2017.1.5)
    • Flutter plugin version 15.1
    • Dart plugin version 171.4694.29

[✓] Connected devices
    • iPhone SE • 464ACEFB-D361-4889-90B9-4084691BDF19 • ios • iOS 10.3 (simulator)

**Comment by InMatrix** [comment]
I upgraded IntelliJ to 2017.2, but it didn't fix the problem.

Version info
------------
[✓] Flutter (on Mac OS X 10.12.5 16F73, locale en-US, channel master)
    • Flutter at /Users/taodong/Code/flutter_repos/flutter
    • Framework revision 9d901327a4 (22 minutes ago), 2017-07-20 17:06:57 -0700
    • Engine revision 5fcfb995bb
    • Tools Dart version 1.25.0-dev.4.0

[✓] Android toolchain - develop for Android devices (Android SDK 25.0.2)
    • Android SDK at /Users/taodong/Library/Android/sdk
    • Platform android-24, build-tools 25.0.2
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_112-release-b06)

[✓] iOS toolchain - develop for iOS devices (Xcode 8.3.3)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 8.3.3, Build version 8E3004b
    • ios-deploy 1.9.1
    • CocoaPods version 1.2.0

[✓] Android Studio (version 2.3)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Java version OpenJDK Runtime Environment (build 1.8.0_112-release-b06)

[✓] IntelliJ IDEA Community Edition (version 2017.2)
    • Flutter plugin version 15.1
    • Dart plugin version 172.3317.48

[✓] Connected devices
    • iPhone SE • 464ACEFB-D361-4889-90B9-4084691BDF19 • ios • iOS 10.3 (simulator)

**Comment by devoncarew** [comment]
Yup, I can confirm. I'm looking at this - I think that IntelliJ 2017.2 may have introduced some delay or buffering when saving changes to disk.

**Comment by devoncarew** [comment]
I did some more investigation - the issue isn't with the recent IntelliJ rev. I bisected the issue to df5cb390e2a44464d2edf499952dca8ed64a238f (/cc @tvolkert  - Rev engine to 488584f8b7cf188d4699880d7b55144cd48067bf #11291). Unfortunately that's an engine roll, so this will take more digging.

**Comment by devoncarew** [comment]
This issue was moved to flutter/flutter#11336

**Comment by devoncarew** [comment]
Here are the engine commits for that range:

```
488584f8b Add `PluginRegistry.Registrar#view()` (#3900)
61c4898a9 Add support for flow traces in fml/trace_event. (#3903)
106118a7d Update Dart revision to a83a69ff3eff845980308779cadbdb80cc03a603. (#3902)
777bb8285 Enable the “Highlight Vsync” option in the timeline viewer on Android. (#3901)
925298d94 Update the content handler to use the Mozart session API. (#3887)
d098b358b Fix Mac desktop platform for use with std::weak_ptr in the PlatformView. (#3897)
de00757dd Revert "Revert "Reland "Run Flutter on iOS and Android with color correct Skia (#3826)" (#3878)" (#3895)
cfa180de5 Use shared library output from shared toolchain (#3869)
569de1e6d Revert "Revert "Switch to new encode API on SkPixelSerializer"" (#3896)
dd2fb3c04 Accumulate a batch of Skia objects that will be destructed later on the IO thread (#3888)
bd2294381 Revert "Switch to new encode API on SkPixelSerializer (#3892)" (#3894)
aabc95df8 Switch to new encode API on SkPixelSerializer (#3892)
ad7f258bf Fix usage of std::labs (#3891)
ab8584eff Roll Skia and enable API guards (#3884)
015e8cf52 Ensure that the source rect used for the the raster cache replacement is the same as the size of the N32Premul surface. (#3889)
```

**Comment by devoncarew** [comment]
I haven't bisected this, but the dart roll is a likely candidate: 106118a7dbf874cda48bf71eafc956d36edaa46f, rolling to dart sdk a83a69ff3eff845980308779cadbdb80cc03a603.
